distributed/tests/test_utils.py

```diff
@@ -149,8 +149,18 @@ def test_get_ip_interface():
         assert get_ip_interface("lo") == "127.0.0.1"
     else:
         pytest.skip("test needs to be enhanced for platform %r" % (sys.platform,))
-    with pytest.raises(KeyError):
-        get_ip_interface("__non-existent-interface")
+
+    non_existent_interface = "__non-existent-interface"
+    expected_error_message = (
+        "{!r} is not a valid network interface.+"
+        "Valid network interfaces are:".format(non_existent_interface)
+    )
+    if sys.platform == "darwin":
+        expected_error_message += ".+'lo0'"
+    elif sys.platform.startswith("linux"):
+        expected_error_message += ".+'lo'"
+    with pytest.raises(KeyError, match=expected_error_message):
```

Comment:
Small nit:

Rather than test that the entire error message is exactly this, we might instead test some properties of it that we care about, like that "network interface" and the names of other interfaces are present in the message.

If someone changes the error message but includes this kind of information I think that that's ok, and shouldn't be considered an error.